src/AppProviders.tsx

```diff
@@ -58,7 +59,9 @@ export function AppProviders({ children }: ProvidersProps) {
                       <MarketDataProvider>
                         <TransactionsProvider>
                           <ModalProvider>
-                            <DefiProvider>{children}</DefiProvider>
+                            <BuySellManagerProvider>
+                              <DefiProvider>{children}</DefiProvider>
```

Comment:
Disclaimer: I didn't start the proper review of this yet, so I might come back to what I said later on, take this with a grain of salt 🧂!

I feared this would potentially be used, since I made the same "mistake" (again, pinch of salt until I have a more in-depth look) while starting work on modals. tl;dr, `<DefiProvider />` is exclusively for the `DeFi` section and shouldn't be used to create modals.
It has very specific routing/logic and overall is a non-conventional modal for us (using a mix of controlled/uncontrolled logic etc).
`useModal()` should be used to consume modals, and their definition should live in`src/context/ModalProvider/ModalProvider.tsx`:
https://github.com/shapeshift/web/blob/e22521102e767406bca821d3dad025ec31bcfc39/src/context/ModalProvider/ModalProvider.tsx#L14

tl;dr 
- We don't need another provider for this, `ModalProvider` should do the trick and using a `<MemoryRouter />` should solve your modal routing needs
- `<DefiProvider />`shouldn't be used for anything else than DeFi context

Reply:
Thanks for your comment!